systems/registry/org/pkg/server/org.go

```diff
@@ -27,21 +31,142 @@ type OrgService struct {
 	userRepo       db.UserRepo
 	orgName        string
 	baseRoutingKey msgbus.RoutingKeyBuilder
+	RegistryUserService client.RegistryUsersClientProvider
 	msgbus         mb.MsgBusServiceClient
 	pushgateway    string
+	notification   client.NotificationClient
 }
 
-func NewOrgServer(orgRepo db.OrgRepo, userRepo db.UserRepo, defaultOrgName string, msgBus mb.MsgBusServiceClient, pushgateway string) *OrgService {
+func NewOrgServer(orgRepo db.OrgRepo, userRepo db.UserRepo, defaultOrgName string, msgBus mb.MsgBusServiceClient, pushgateway string, notification client.NotificationClient ,RegistryUserService client.RegistryUsersClientProvider) *OrgService {
 	return &OrgService{
 		orgRepo:        orgRepo,
 		userRepo:       userRepo,
 		orgName:        defaultOrgName,
 		baseRoutingKey: msgbus.NewRoutingKeyBuilder().SetCloudSource().SetContainer(pkg.ServiceName),
 		msgbus:         msgBus,
+		RegistryUserService: RegistryUserService,
 		pushgateway:    pushgateway,
+		notification:   notification,
 	}
 }
 
+func (o *OrgService) AddInvitation(ctx context.Context, req *pb.AddInvitationRequest) (*pb.AddInvitationResponse, error) {
+	log.Infof("Adding invitation %v", req)
+
+	if req.GetOrg() == "" {
+		return nil, status.Errorf(codes.InvalidArgument, "Org is required")
+	}
+
+	if req.GetEmail() == "" {
+		return nil, status.Errorf(codes.InvalidArgument, "Email is required")
+	}
+
+	if req.GetName() == "" {
+		return nil, status.Errorf(codes.InvalidArgument, "Name is required")
+	}
+
+	link, err := generateInvitationLink()
+	if err != nil {
+		return nil, err
+	}
+	expiresAt := time.Now().Add(3 * 24 * time.Hour) // Set the expiration time to three days from now
+
+	invitationId := uuid.NewV4()
+	err = o.orgRepo.AddInvitation(
+		&db.Invitation{
+			Id:        invitationId,
+			Org:       req.GetOrg(),
+			Name:      req.GetName(),
+			Link:      link,
+			Email:     req.GetEmail(),
+			Role:      pbRoleTypeToDb(req.GetRole()),
+			ExpiresAt: expiresAt,
+			Status:    db.InvitationStatus(req.GetStatus()),
+		},
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	res, err := o.orgRepo.GetByName(req.GetOrg())
+	if err != nil {
+		return nil, err
+	}
+
+	userRegistrySvc, err := o.RegistryUserService.GetClient()
```

Comment:
```go
userRegistrySvc, err := o.RegistryUserService.GetClient()
	if err != nil {
		return nil, err
	}
	remoteUserResp, err := userRegistrySvc.Get(ctx,
		&userRegpb.GetRequest{UserId: res.Owner.String()})
	if err != nil {
		return nil, err
	}
```
I think it is even better to do all these before adding the invitatin into DB.

Reply:
Yeah make sense.